posthog/api/cohort.py

```diff
@@ -168,7 +168,7 @@ def persons(self, request: Request, **kwargs) -> Response:
             filter = filter.with_data({LIMIT: 100})
 
         raw_result = sync_execute(
-            GET_COHORTPEOPLE_BY_COHORT_ID,
+            GET_STATIC_COHORTPEOPLE_BY_COHORT_ID if cohort.is_static else GET_COHORTPEOPLE_BY_COHORT_ID,
```

Comment:
Don't we have most of this code in ee/clickhouse/models/cohort.py already? We're duplicating the business logic into two separate places now, making further changes harder/cause bugs.

Reply:
`ee/clickhouse/models/cohort.py` houses filtering logic. There is a slight bit of repetition in some cases where the cohort subquery selects from these tables. 

This logic is purely retrieval based and not nested in a subquery to filter. It's new because previously the way we were retrieving persons on a cohort was solely from postgres using a cohort_id. Since we're only selectively storing cohort relationships in postgres now, the retrieval needs to be from clickhouse